solver/src/settlement.rs

```diff
@@ -111,26 +117,61 @@ impl Settlement {
         Ok(cursor.into_inner())
     }
 
-    // For now this computes the total surplus of all EOA trades.
-    pub fn objective_value(&self) -> U256 {
-        // TODO: in order to achieve a fair comparison between different solutions, we need to normalize the price
-        // vector as otherwise any price vector scaling by factor x would also scale the objective value by that factor
-        // without affecting the validity of the solution.
-        match self.trades.iter().fold(Some(U256::zero()), |acc, trade| {
-            let sell_token_price = self
-                .clearing_prices
+    fn total_surplus(&self, prices: &HashMap<H160, BigRational>) -> Option<BigRational> {
+        self.trades.iter().fold(Some(num::zero()), |acc, trade| {
+            let sell_token_external_price = prices
                 .get(&trade.order.sell_token)
                 .expect("Solution with trade but without price for sell token");
-            let buy_token_price = self
-                .clearing_prices
+            let buy_token_external_price = prices
                 .get(&trade.order.buy_token)
-                .expect("Solution with trade but without price for sell token");
-            acc?.checked_add(trade.surplus(*sell_token_price, *buy_token_price)?)
-        }) {
+                .expect("Solution with trade but without price for buy token");
+            acc?.checked_add(&trade.surplus(sell_token_external_price, buy_token_external_price)?)
+        })
+    }
+
+    // Objective is re-computed using external prices.
+    fn objective_value_v1(
+        &self,
+        external_prices: &HashMap<H160, BigRational>,
+    ) -> Option<BigRational> {
+        self.total_surplus(external_prices)
+    }
+
+    // Objective is scaled by (harmonic) mean of external/found price ratio.
+    #[allow(dead_code)]
```

Comment:
I think we should check this in on a separate branch/commit but not have dead_code in the main codebase.

Reply:
Removed v2 here: f97dd2c.